exchange/src/exchange/bots/bot_types/retail.hpp

```diff
@@ -1,27 +1,33 @@
 #pragma once
 
-#include "generic_bot.hpp"
+#include "exchange/traders/trader_types/bot_trader.hpp"
 #include "shared/messages_wrapper_to_exchange.hpp"
 
 namespace nutc {
 
 namespace bots {
 
-class RetailBot : public GenericBot {
+class RetailBot : public BotTrader {
     std::random_device rd{};
     std::mt19937 gen{};
     std::poisson_distribution<> poisson_dist{};
 
 public:
-    RetailBot(std::string bot_id, double interest_limit) :
-        GenericBot(std::move(bot_id), interest_limit),
+    RetailBot(std::string ticker, double interest_limit) :
+        BotTrader(std::move(ticker), interest_limit),
         AGGRESSIVENESS(std::normal_distribution<>{50, 2000}(gen))
-    {}
+    {
+        set_capital(interest_limit);
+    }
 
     [[nodiscard]] bool is_active() const override;
 
     std::optional<messages::MarketOrder> take_action(double current, double theo);
 
+    RetailBot(RetailBot&& other) noexcept :
```

Comment:
put all the ctors together

also why is agressiveness in all caps its not a compile time const

Reply:
because its a const class member